daemon/daemon.go

```diff
@@ -105,7 +106,10 @@ type Daemon struct {
 	seccompProfile     []byte
 	seccompProfilePath string
 
-	usage singleflight.Group
+	usageContainers ctxsingleflight.Group[string, []*types.Container]
+	usageImages     ctxsingleflight.Group[string, []*types.ImageSummary]
+	usageVolumes    ctxsingleflight.Group[string, []*volume.Volume]
+	usageLayer      ctxsingleflight.Group[string, int64]
```

Comment:
Given that only one constant "key" is used for each group respectively, I think it would make the code more self-documenting to use a key type with only one possible value: `struct{}`.

Reply:
Good point, thanks!